### ISSUE TITLE
read model information

### ISSUE THREAD

**Issue by Arnold1** [google/yggdrasil-decision-forests#40]
hi,

i have a model with a /assets folder:
- 725b7a901c7141c3data_spec.pb
- 725b7a901c7141c3done
- 725b7a901c7141c3header.pb
- 725b7a901c7141c3nodes-00000-of-00001
- 725b7a901c7141c3random_forest_header.pb

im trying to read some infos about the generated model:
```
package main

import (
	"log"
	"fmt"
	"github.com/google/yggdrasil-decision-forests/yggdrasil_decision_forests/port/go/model/io"
	"github.com/google/yggdrasil-decision-forests/yggdrasil_decision_forests/port/go/serving"
	gbt "github.com/google/yggdrasil-decision-forests/yggdrasil_decision_forests/port/go/model/randomforest"
)

func main() {
	modelPath := "../foo/tf-model/assets"
	model, err := io.LoadModel(modelPath)
	if err != nil {
		log.Fatalf("Cannot load model. %v", err)
	}

	rfModel := model.(*gbt.Model)

	varImportance := rfModel.Header().GetPrecomputedVariableImportances()
	for k,v := range varImportance {
		log.Printf("%s, %s\n", k, v)
	}

	// Compile the model for fast infernce.
	// At this point, the "model objet can be discarded.
	engine, err := serving.NewEngine(model)
	if err != nil {
		log.Fatalf("Cannot create serving.NewEngine. %v", err)
	}

	//fmt.Printf("\n%+v\n", engine.Features().NumericalFeatures)
	//fmt.Printf("\n%+v\n", engine.Features().CategoricalFeatures)
	//fmt.Printf("%+v\n", engine.Features().CategoricalSpec)
}
```

i get this output. how to interprete the information about the feature importance? is there other useful information about the feature importance or something else i can read?
```
2023/04/12 12:38:00 NUM_AS_ROOT, variable_importances:{attribute_idx:10 importance:1}
2023/04/12 12:38:00 MEAN_MIN_DEPTH, variable_importances:{attribute_idx:3 importance:23.4122039822913} variable_importances:{attribute_idx:14 importance:23.4122039822913} variable_importances:{attribute_idx:13 importance:18.959115431559333} variable_importances:{attribute_idx:9 importance:18.90570766197468} variable_importances:{attribute_idx:7 importance:18.404960169864875} variable_importances:{attribute_idx:6 importance:16.509970172537066} variable_importances:{attribute_idx:4 importance:15.318525526674996} variable_importances:{attribute_idx:12 importance:13.434679300607382} variable_importances:{attribute_idx:8 importance:11.116124885348432} variable_importances:{attribute_idx:2 importance:10.598752013173195} variable_importances:{attribute_idx:0 importance:8.497324194911275} variable_importances:{attribute_idx:5 importance:8.410723442363665} variable_importances:{attribute_idx:1 importance:6.79472494457003} variable_importances:{attribute_idx:11 importance:6.022468096170096} variable_importances:{attribute_idx:10 importance:0}
2023/04/12 12:38:00 NUM_NODES, variable_importances:{attribute_idx:7 importance:28824} variable_importances:{attribute_idx:4 importance:24230} variable_importances:{attribute_idx:12 importance:23355} variable_importances:{attribute_idx:9 importance:16055} variable_importances:{attribute_idx:5 importance:15593} variable_importances:{attribute_idx:11 importance:12692} variable_importances:{attribute_idx:10 importance:4472} variable_importances:{attribute_idx:6 importance:3964} variable_importances:{attribute_idx:13 importance:3655} variable_importances:{attribute_idx:8 importance:2105} variable_importances:{attribute_idx:1 importance:1939} variable_importances:{attribute_idx:2 importance:1360} variable_importances:{attribute_idx:0 importance:218}
2023/04/12 12:38:00 SUM_SCORE, variable_importances:{attribute_idx:5 importance:33101.01888473334} variable_importances:{attribute_idx:11 importance:22825.741772663732} variable_importances:{attribute_idx:7 importance:21561.736182725967} variable_importances:{attribute_idx:12 importance:21390.187111173334} variable_importances:{attribute_idx:4 importance:21388.94658304365} variable_importances:{attribute_idx:10 importance:19679.342864750575} variable_importances:{attribute_idx:9 importance:13148.756573859178} variable_importances:{attribute_idx:1 importance:6424.683870845609} variable_importances:{attribute_idx:6 importance:5866.794061057355} variable_importances:{attribute_idx:2 importance:4778.448671128921} variable_importances:{attribute_idx:8 importance:4520.945368736043} variable_importances:{attribute_idx:13 importance:4238.837772123543} variable_importances:{attribute_idx:0 importance:1949.4544305546806}
```

is there a way to generate some plot like the following based on the info above? https://shap.readthedocs.io/en/latest/example_notebooks/tabular_examples/tree_based_models/Scatter%20Density%20vs.%20Violin%20Plot%20Comparison.html#Layered-violin-plot

also tried to read the HyperparameterOptimizerLogs - but it seems empty - getting nil:
```
fmt.Printf("%+v\n", rfModel.Header().GetHyperparameterOptimizerLogs())
```

Thanks

**Comment by rstz** [comment]
Hi,

There's more than one question in this issue, could you please split all your questions into separate issues? This really helps readers and it also helps the team triage open questions.

**Comment by Arnold1** [comment]
@rstz sure - changed it. will open a second issue for the other question.

**Comment by janpfeifer** [comment]
hi @Arnold1 ,

"Variable Importance" is typically used as a proxy metric to how useful a feature is. This can be handy to help debug a model (if a model is unexpectedly important it may be leaking the label somehow) or to select features, if they are expensive to generate.

There are various definitions of variable importance: e.g: "NUM_AS_ROOT" is the number of times a feature was used in root node of trees in a random forest. "MEAN_MIN_DEPTH" is the mean minimum depth a feature appears in the trees: the closer to the root, so the smaller the value, the more important the feature.

This is mentioned in the [TF-DF tutorial on inspecting the model](https://www.tensorflow.org/decision_forests/tutorials/advanced_colab), but I couldn't find a reference documentation on all available variable importance definitions.

@rstz , do you know if there is a documentation page with the list and description of Variable Importance metrics ?

cheers
